crates/client/l1-messages/src/config.rs

```diff
@@ -0,0 +1,36 @@
+use std::fs::File;
+use std::path::PathBuf;
+
+pub const DEFAULT_ETHEREUM_NODE: &str = "127.0.0.1:8545";
+pub const DEFAULT_CONTRACT_ADDRESS: &str = "0x5fbdb2315678afecb367f032d93f642f64180aa3";
+
+#[derive(Clone, PartialEq, serde::Deserialize, Debug)]
+pub struct L1MessagesWorkerConfig {
+    #[serde(default = "default_http")]
+    pub http_provider: String,
```

Comment:
You should probably use the `Url` type rather than a string there. Unless there is some reason not to downstream.

Reply:
`ethers` api prefers String.

Reply:
https://docs.rs/ethers/2.0.11/ethers/providers/struct.Http.html#method.new
No they seems to be encouraging the use of `Url`